Documentation/gettingstarted/local-redirect-policy.rst

```diff
@@ -301,5 +309,147 @@ configuring the CiliumLocalRedirectPolicy.
+traffic from application pods to the cluster's DNS service VIP by default, which will be
+bypassed when Cilium is handling service resolution at or before the veth interface of the
+application pod. To enable the DNS node-cache in a Cilium cluster, the following example
+steers traffic to a local DNS node-cache which runs as a normal pod.
+
+* Deploy DNS node-cache in pod namespace.
+
+  .. tabs::
+
+    .. group-tab:: Quick Deployment
+
+        Deploy DNS node-cache.
+
+        .. note::
+
+           The quick deployment example yaml deploys node-local-dns in ``kube-system`` namespace,
+           this does not work for GKE clusters.
+
+        .. parsed-literal::
+
+            $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes-local-redirect/node-local-dns.yaml
+
+    .. group-tab:: Manual Configuration
+
+         * Make sure to use a Node-local DNS image with a release version >= 1.15.16.
+           This is to ensure that we have a knob to disable dummy network interface creation/deletion in
+           Node-local DNS when we deploy it in non-host namespace.
+
+         * Modify Node-local DNS cache's deployment yaml to pass these additional arguments to node-cache:
+           ``-skipteardown=true``, ``-setupinterface=false``, and ``-setupiptables=false``.
+
+         * Modify Node-local DNS cache's deployment yaml to put it in non-host namespace by setting
+           ``hostNetwork: false`` for the daemonset.
+
+         * In the Corefile, bind to ``0.0.0.0`` instead of the static IP.
+
+         * In the Corefile, let CoreDNS serve health-check on its own IP instead of the static IP by
+           removing the host IP string after health plugin.
+
+         * Modify Node-local DNS cache's deployment yaml to point readiness probe to its own IP by
+           removing the ``host`` field under ``readinessProbe``.
+
+* Deploy local redirect policy (LRP) to steer DNS traffic to the node local dns cache.
+
+  .. parsed-literal::
+
+      $ kubectl apply -f \ |SCM_WEB|\/examples/kubernetes-local-redirect/node-local-dns-lrp.yaml
+
+  .. note::
+
+      * The LRP above uses ``kube-dns`` for the cluster DNS service, however if your cluster DNS service is different,
+        you will need to modify this example LRP to specify it.
+      * The namespace specified in the LRP above is set to the same namespace
+        as the node-local DNS cache daemonset.
+      * The LRP above uses the same port names ``dns`` and ``dns-tcp`` as the example quick deployment yaml, you will
+        need to modify those to match your deployment if they are different.
+
+After all ``node-local-dns`` pods are in ready status, DNS traffic should now go to the local node-cache first.
+You can verify by  running ``cilium monitor`` with ``related-to`` set to the endpoint id of one of the
+application pods, or by checking the DNS cache's metrics such as ``coredns_cache_hits_total`` and
+``coredns_cache_misses_total`` via curling ``<node-local-dns-pod>:9253/metrics``.
```

Comment:
Just to be clear, what is the user checking for? That these are nonzero?

Reply:
Added one more sentence after this to explain what should be seen.